server/utils/applications/withdrawables/index.test.ts

```diff
@@ -130,3 +134,59 @@ describe('withdrawableTypeRadioOptions', () => {
     })
   })
 })
+
+describe('sortWithdrawables', () => {
+  const { endDate } = withdrawableFactory.build().dates[0]
+
+  it('should sort withdrawables based on start date', () => {
+    const withdrawables = [
+      withdrawableFactory.build({
+        dates: [{ startDate: '2022-01-01', endDate }],
+      }),
+      withdrawableFactory.build({
+        dates: [{ startDate: '2022-02-01', endDate }],
+      }),
+      withdrawableFactory.build({
+        dates: [{ startDate: '2022-03-01', endDate }],
+      }),
+    ]
+
+    const sortedWithdrawables = sortWithdrawables(withdrawables)
+
+    expect(sortedWithdrawables).toEqual([withdrawables[0], withdrawables[1], withdrawables[2]])
+  })
+
+  it('should handle withdrawables without start date', () => {
+    const withdrawables = [
+      withdrawableFactory.build({ dates: [{ startDate: '2022-01-01', endDate }] }),
+      withdrawableFactory.build({ dates: [] }),
+      withdrawableFactory.build({ dates: [{ startDate: '2022-03-01', endDate }] }),
+    ]
+
+    const sortedWithdrawables = sortWithdrawables(withdrawables)
+
+    expect(sortedWithdrawables).toEqual([withdrawables[0], withdrawables[1], withdrawables[2]])
+  })
```

Comment:
I'm a bit confused here about exactly how we're handling bookings without start date. In the commit message you say:

> we will put the withdrawable without a date first

but this scenario shows it in between items with valid start dates. Again perhaps this could be clearer, with:

- a more explicit scenario title which states exactly how an item without start date will be positioned
- test fixtures which are set up in an order which differs to their final sorted order

Reply:
We now throw an error if there is no start date for a withdrawable so I have updated this test